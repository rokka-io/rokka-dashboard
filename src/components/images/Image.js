import React, { PropTypes } from 'react'
import cx from 'classnames'

const Image = (props) => {
  const onClick = (e) => {
    if (props.onClick) {
      props.onClick(e, props)
    }
  }

  const classes = cx(
    'rka-image-tile',
    props.className,
    {
      'is-uploaded': props.uploaded,
      'is-uploading': !props.uploaded,
      'is-clickable': props.onClick
    }
  )

  const specs = [
    props.format ? props.format.toUpperCase() : null,
    props.size ? `${Math.round(props.size / 1024)}KB` : null,
    [props.width || '?', props.height || '?'].join('Ã—')
  ].filter(line => line).join(', ')

  return (
    <div className="col-md-3 col-sm-6 col-xs-12">
      <div className={classes}
        onClick={onClick}>
        <div className="rka-image-tile-hd">
          {props.uploaded === true &&
            <span className="rka-image-checkbox-container">
              <i className="roka-image-checkbox rka-checkbox-icon is-checked" />
            </span>
          }
          <img src={props.url} alt={props.name} />
        </div>
        <div className="rka-image-tile-ft">
          <div className="rka-image-tile-specs txt-ellipsis" title={specs}>
            {specs}
          </div>
          <div className="rka-image-tile-name txt-ellipsis" title={props.name}>{props.name}</div>
        </div>
      </div>
    </div>
  )
}

Image.propTypes = {
  url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  width: PropTypes.number,
  height: PropTypes.number,
  size: PropTypes.number,
  format: PropTypes.string,
  uploaded: PropTypes.bool,
  onClick: PropTypes.func,
  className: PropTypes.string
}

export default Image
