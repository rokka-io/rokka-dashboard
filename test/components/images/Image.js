import React from 'react'
import { shallow } from 'enzyme'
import Image from '../../../src/components/images/Image'
import renderer from 'react-test-renderer'

test('Image does render with minimal props', () => {
  const component = renderer.create(
    <Image url="http://example.org/image.png" name="Example name" />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Image does render with all props', () => {
  const component = renderer.create(
    <Image
      url="http://example.org/image.png"
      name="Example name"
      width={100}
      height={100}
      size={10000}
      format="png"
      uploaded
      onClick={() => {
        throw new Error('Should not be called')
      }}
      className="additional-classes"
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Image onError triggers an error', () => {
  const component = shallow(
    <Image
      url="http://example.org/image.png"
      name="Example name"
      width={100}
      height={100}
      size={10000}
      format="png"
      uploaded
      onClick={() => {
        throw new Error('Should not be called')
      }}
      className="additional-classes"
    />
  )

  expect(component.find('img').exists()).toBe(true)

  component.setState({imageError: true})

  expect(component.find('img').exists()).toBe(false)
})
