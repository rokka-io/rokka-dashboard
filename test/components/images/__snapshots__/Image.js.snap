// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image does render with all props 1`] = `
<div
  className="col-md-3 col-sm-6 col-xs-12"
>
  <div
    className="rka-image-tile additional-classes is-uploaded is-clickable"
    onClick={[Function]}
  >
    <div
      className="rka-image-tile-hd"
    >
      <a
        className="rka-image-tile-link rka-link"
        href="http://example.org/image.png"
        onClick={[Function]}
        target="_blank"
      >
        Open in new window
      </a>
      <span
        className="rka-image-checkbox-container"
      >
        <i
          className="roka-image-checkbox rka-checkbox-icon is-checked"
        />
      </span>
      <img
        alt="Example name"
        onError={[Function]}
        src="http://example.org/image.src.png"
      />
    </div>
    <div
      className="rka-image-tile-ft"
    >
      <div
        className="rka-image-tile-specs txt-ellipsis"
        title="PNG, 10KB, 100×100"
      >
        PNG, 10KB, 100×100
      </div>
      <div
        className="rka-image-tile-name txt-ellipsis"
        title="Example name"
      >
        Example name
      </div>
    </div>
  </div>
</div>
`;

exports[`Image does render with minimal props 1`] = `
<div
  className="col-md-3 col-sm-6 col-xs-12"
>
  <div
    className="rka-image-tile is-uploading"
    onClick={[Function]}
  >
    <div
      className="rka-image-tile-hd"
    >
      <a
        className="rka-image-tile-link rka-link"
        href="http://example.org/image.png"
        onClick={[Function]}
        target="_blank"
      >
        Open in new window
      </a>
      <img
        alt="Example name"
        onError={[Function]}
        src="http://example.org/image.src.png"
      />
    </div>
    <div
      className="rka-image-tile-ft"
    >
      <div
        className="rka-image-tile-specs txt-ellipsis"
        title="?×?"
      >
        ?×?
      </div>
      <div
        className="rka-image-tile-name txt-ellipsis"
        title="Example name"
      >
        Example name
      </div>
    </div>
  </div>
</div>
`;
